scripts/imgtool/image.py

```diff
@@ -111,7 +111,7 @@ def add(self, kind, payload):
         """
         e = STRUCT_ENDIAN_DICT[self.endian]
         if isinstance(kind, int):
-            buf = struct.pack(e + 'BBH', kind, 0, len(payload))
+            buf = struct.pack(e + 'HH', kind, len(payload))
```

Comment:
Makes sense. For correctness with the description in https://github.com/mcu-tools/mcuboot/blob/main/boot/bootutil/include/bootutil/image.h#L100, I suggest you add code to test the value is between `0x00a0` and `0xfffe`. If it's not you can: `raise click.UsageError("Invalid TLV value")`

Reply:
@utzig thanks for your comment and suggestion but as far as I can see, the reserved range for vendors isn't continuous, e.g. 0x00a0 - 0x00ff, 0x01a0 - 0x01ff ... Wouldn't a simple 0x00a0 ~ 0xfffe range test be misleading in that case? Or am I missing something here?

Reply:
I think that's just convention; it should also work for something like 0x0100 as far as I can tell. What's more important is to not accept 0xffff and values already defined.

Reply:
OK, I've added range validity check, updated commit message, title and rebased on latest main.

The overlapping with predefined values is checked earlier: https://github.com/mcu-tools/mcuboot/blob/main/scripts/imgtool/main.py#L394